### PR TITLE
enable pip and skimage.data caching on Azure

### PR DESCRIPTION
**Pull request by grlee77**
## Description

This PR enables some caching of downloaded files on Azure as well as some other minor modifications. This was split of from #5338 where the caching behavior was originally verified. That PR will focus only on the change to the face classification example while this one will have the CI changes.

The goal here is to help avoid the somewhat common timeouts that have been seen recently in our Azure CI jobs involving either the documentation or gallery example builds.

## Checklist

- [Docstrings for all functions](https://github.com/numpy/numpy/blob/master/doc/example.py)
- Gallery example in `./doc/examples` (new features only)
- Benchmark in `./benchmarks`, if your changes aren't covered by an
  existing benchmark
- Unit tests
- Clean style in [the spirit of PEP8](https://www.python.org/dev/peps/pep-0008/)

## For reviewers

- Check that the PR title is short, concise, and will make sense 1 year
  later.
- Check that new functions are imported in corresponding `__init__.py`.
- Check that new features, API changes, and deprecations are mentioned in
      `doc/release/release_dev.rst`.

### CODE REVIEW

**Comment by grlee77** [comment]
If this looks good to others, we can open an issue to enable similar caching on GitHub Actions.

File: azure-pipelines.yml [hunk 1]
```diff
@@ -48,121 +51,146 @@ jobs:
...
+      displayName: Cache pip packages
+
+    - bash: |
+        set -ex
+        PYTHON="$(python.pythonLocation)\\python.exe"
+
+        # Update pip
+        $PYTHON -m pip install -U pip setuptools wheel
+        # print out the pip cache dir followed by the variable defined above
+	# to confirm that they match
```
Comment (jni) [inline azure-pipelines.yml:84]:
```suggestion
	    # to confirm that they match
```

**Review by jni — COMMENTED** [review]
@grlee77 the diff is pretty huge, can you point out the relevant bits? I expect you didn't actually change every line?

**Comment by grlee77** [comment]
> @grlee77 the diff is pretty huge, can you point out the relevant bits? I expect you didn't actually change every line?

If you look at the first three commits individually it is much smaller. 

The last commit is only indentation and can be dropped (at one point I saved in VS Code while using an Azure yml validator plugin and it made these changes, but they are not truly necessary).

**Review by jni — APPROVED** [review]

**Comment by jni** [comment]
Great, thanks, that was much easier! :sweat_smile: 

There's one minor indentation fix remaining, otherwise LGTM. I'll merge tomorrow if there's no further comments.

**Comment by grlee77** [comment]
I modified the last commit and the indentation should be consistent now. I think if you look at the Azure CI this time around you will see things being retrieved from the caches.

**Comment by grlee77** [comment]
Thanks for reviewing @jni. I went ahead and merged to see if we end up getting fewer CI failures in practice

**Pull request merged**